vendor/engines/secure_rooms/app/services/secure_rooms/access_handlers/occupancy_handler.rb

```diff
@@ -17,22 +17,30 @@ def initialize(event)
       end
 
       def process
+        Rails.logger.info("[SecureRooms] Entered SecureRooms::AccessHandlers::EventHandler#process")
+
         return unless event.success?
 
         if current_occupant? && exiting?
+          Rails.logger.info("[SecureRooms] Processing branch current_occupant? && exiting? (existing occupancy id: #{existing_occupancy.id})")
           existing_occupancy.associate_exit!(event)
         elsif new_occupant? && entering?
+          Rails.logger.info("[SecureRooms] Processing branch new_occupant? && entering? (new occupancy id: #{new_occupancy.id})")
           new_occupancy.associate_exit!(event) if entry_only?
           new_occupancy.associate_entry!(event)
         elsif new_occupant? && exiting?
+          Rails.logger.info("[SecureRooms] Processing branch new_occupant? && exiting? (new occupancy id: #{new_occupancy.id})")
           new_occupancy.mark_orphaned!
           new_occupancy.associate_exit!(event)
         elsif current_occupant? && entering?
+          Rails.logger.info("[SecureRooms] Processing branch current_occupant? && entering? (existing occupancy id: #{existing_occupancy.id})")
           complete_existing_occupancy!
           new_occupancy.associate_entry!(event)
         else
           raise NotImplementedError, "Encountered unexpected scan context"
         end
+
+        Rails.logger.info("[SecureRooms] Exiting SecureRooms::AccessHandlers::EventHandler#process")
```

Comment:
I'm pretty sure this `process` method needs to return an `occupancy`. It's probably not clear, but the `Occupancy#associate_entry!` methods returns `self`.

Reply:
Yeah, that was very surprising! I have a fix incoming for this, just read it after fixing the problem 😄